Library/Homebrew/cmd/vendor-install.sh

```diff
@@ -119,9 +119,18 @@ fetch() {
     --remote-time
     --location
     --user-agent "${HOMEBREW_USER_AGENT_CURL}"
-    --header "Authorization: Bearer ${HOMEBREW_DOCKER_REGISTRY_TOKEN:-QQ==}"
   )
 
+  if [[ -n "${HOMEBREW_DOCKER_REGISTRY_TOKEN}" ]]
+  then
+    curl_args[${#curl_args[*]}]="--header \"Authorization: Bearer ${HOMEBREW_DOCKER_REGISTRY_TOKEN}\""
+  elif [[ -n "${HOMEBREW_DOCKER_REGISTRY_BASIC_AUTH_TOKEN}" ]]
+  then
+    curl_args[${#curl_args[*]}]="--header \"Authorization: Basic ${HOMEBREW_DOCKER_REGISTRY_BASIC_AUTH_TOKEN}\""
+  else
+    curl_args[${#curl_args[*]}]="--header \"Authorization: Bearer QQ==\""
+  fi
```

Comment:
```suggestion
  if [[ -z "${HOMEBREW_DOCKER_REGISTRY_TOKEN}" ]]
  then
    HOMEBREW_DOCKER_REGISTRY_TOKEN="${HOMEBREW_DOCKER_REGISTRY_BASIC_AUTH_TOKEN:-QQ==}"
  fi
```
so that we can move this before `curl_args+=(` and then just refer to `HOMEBREW_DOCKER_REGISTRY_TOKEN` in the argument to `--header`.

Reply:
That said, I don't understand why you don't just set whatever is in `HOMEBREW_DOCKER_REGISTRY_BASIC_AUTH_TOKEN` also to `HOMEBREW_DOCKER_REGISTRY_TOKEN`.

Reply:
`Bearer` vs `Basic` is the important distinction here.

Reply:
Aha yes, I see. My mistake!